### ISSUE TITLE
[TD-123] Taskd not syncing when "127.0.0.1 <servername>" is set in /etc/host

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskserver#123]
_Sebastian Sonne on 2016-06-04T22:45:36Z says:_ 

 For something on my webserver, I've set "127.0.0.1 <local hostname>" in /etc/hosts. From that point on, I could not sync with taskd anymore, although taskd itself started up fine and reported no errors. Removing the line solved the issue again.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2016-06-04T22:45:36Z
Modified: 2016-06-05T08:25:44Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-06-04T23:23:45Z says:_ 

 What was the CN set to in the certs you created?

**Comment by taskwarrior** [comment]
_Sebastian Sonne on 2016-06-05T08:25:44Z says:_ 

 The name of the server (that is, what <servername> in the hosts file is).

**Comment by Triavanicus** [comment]
I tried all of my devices IP addresses, of note `::1` and `127.0.0.1`, on other devices `192.168.x.x` and `fe80*` (which are the device's local network addresses) did not work, and only my public IPv6 address worked.

So it appears that it only listens on the device's public address. This has to do with setting the `server` setting in taskserver's `config` file to the server name, which likely queries the IP address that is found. This was fixed to allow any IPv4 connection with `0.0.0.0` instead. Oddly I was able to allow any ip address (at least with my local clients) by setting the `server` config setting to `server=:::53589`, though that may be due to some IPv4 to IPv6 address conversion/translation.

Updating the doc to explain that the `server` setting is about IP address, and not server name, could help explain to people the purpose of ip, since setting a bind address is common in other server documentation.

So I would also say that this is not a bug, but configuration problem. Allowing people to bind to multiple addresses could be a nice feature though, ex: `server=127.0.0.1:53589,192.168.1.52:53589,ServerName:53589`
